tutorials/xr/openxr_composition_layers.rst

```diff
@@ -0,0 +1,292 @@
+.. _doc_openxr_composition_layers:
+
+OpenXR composition layers
+=========================
+
+Introduction
+------------
+
+In XR games you generally want to create user interactions that happen in 3D space 
+and involve users touching objects as if they are touching them in real life.
+
+Sometimes however it is unavoidable to create a more traditional 2D interface.
+In XR however you can't just add 2D components to your scene.
+Godot needs depth information to properly position these elements so they appear at
+a comfortable place for the user.
+Even with depth information there are headsets with slanted displays that make it impossible 
+for the standard 2D pipeline to correctly render the 2D elements.
+
+The solution then is to render the UI to a :ref:`SubViewport <class_subviewport>`
+and display the result of this using a :ref:`ViewportTexture <class_viewporttexture>` on a 3D mesh.
+The :ref:`QuadMesh <class_quadmesh>` is a suitable option for this.
+
+.. note::
+    See the `GUI in 3D <https://github.com/godotengine/godot-demo-projects/tree/master/viewport/gui_in_3d>`_
+    example project for an example of this approach.
+
+The problem with displaying the viewport in this way is that the rendered result
+is sampled for lens distortion by the XR runtime and the resulting quality loss 
+can make UI text hard to read.
+
+OpenXR offers a solution to this problem through composition layers.
+With composition layers it is possible for the contents of a viewport to be projected
+on a surface after lens distortion resulting in a much higher quality end result.
+
+.. note::
+    As not all XR runtimes support all composition layer types,
+    Godot implements a fallback solution where we render the viewport
+    as part of the normal scene but with the aforementioned quality
+    limitations.
+
+.. warning::
+    When the composition layer is supported,
+    it is the XR runtime that presents the subviewport.
+    This means the UI is only visible in the headset,
+    it will not be accessible by Godot and will thus
+    not be shown when you have a spectator view on the desktop.
+
+There are currently 3 nodes that expose this functionality:
+
+- :ref:`OpenXRCompositionLayerCylinder <class_OpenXRCompositionLayerCylinder>` shows the contents of the SubViewport on the inside of a cylinder.
+- :ref:`OpenXRCompositionLayerEquirect <class_OpenXRCompositionLayerEquirect>` shows the contents of the SubViewport on a warped rectangle.
```

Comment:
I think describing an equirect as the interior of sphere makes it easiest to understand:

```suggestion
- :ref:`OpenXRCompositionLayerEquirect <class_OpenXRCompositionLayerEquirect>` shows the contents of the SubViewport on the interior of a sphere (or "slice" of a sphere).
```

Reply:
I just copied these descriptions from the class definition, so I think if we change it here, we should also change it on the class?

Reply:
Oops, probably my fault then. :-) Yeah, we should update it there as well.

Reply:
I'll adjust it here and then we can update the documentation later.